src/rdf4cpp/rdf/datatypes/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.17)
+
+add_library(rdf4cpp_datatypes OBJECT
+        owl/Rational.cpp
+        owl/Real.cpp
+        rdf/LangString.cpp
+        xsd/integers/non_positive/NegativeInteger.cpp
+        xsd/integers/non_positive/NonPositiveInteger.cpp
+        xsd/integers/non_negative/NonNegativeInteger.cpp
+        xsd/integers/non_negative/PositiveInteger.cpp
+        xsd/integers/non_negative/UnsignedByte.cpp
+        xsd/integers/non_negative/UnsignedInt.cpp
+        xsd/integers/non_negative/UnsignedLong.cpp
+        xsd/integers/non_negative/UnsignedShort.cpp
+        xsd/integers/non_positive/NegativeInteger.cpp
+        xsd/integers/non_positive/NonPositiveInteger.cpp
+        xsd/integers/signed/Byte.cpp
+        xsd/integers/signed/Int.cpp
+        xsd/integers/signed/Integer.cpp
+        xsd/integers/signed/Long.cpp
+        xsd/integers/signed/Short.cpp
+        xsd/Boolean.cpp
+        xsd/Decimal.cpp
+        xsd/Double.cpp
+        xsd/Float.cpp
+        xsd/String.cpp
+        )
+
+target_include_directories(
+        rdf4cpp_datatypes
+        PRIVATE
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>/../../.."
+        )
+
+target_link_libraries(rdf4cpp_datatypes
+        PUBLIC
```

Comment:
can be made private. We don't have any public headers on this target so we also do not depend on any dependent headers to be visible.

Reply:
Oh, is that the only thing `PUBLIC` means in this context I thought it also had something to do with exporting symbols in the binary, good to know